lib/functions/get_pf_medication_validation_data.R

```diff
@@ -79,18 +80,61 @@ get_nhsbsa_data <- function(dataset_id, sql, start_date = NULL, end_date = NULL)
   base_endpoint <- "https://opendata.nhsbsa.net/api/3/action/"
   action_method <- "datastore_search_sql?"
 
+  # Retrieve table names for specified dataset and data range
   table_names <- get_dataset_table_names(dataset_id, start_date, end_date)$table_name
 
+  # Construct URLs for querying the API, one URL per table, modifying each table with table_name
   async_api_calls <- paste0(
     base_endpoint,
     action_method,
     "resource_id=", table_names,
     "&sql=", URLencode(map_chr(table_names, construct_sql_query, sql))
   )
 
-  responses <- crul::Async$new(urls = async_api_calls)$get()
+  # Initialise an empty list to store API responses
+  responses <- list()
+
+  # Loop over each API call URL, by:
+  # 1) getting URL for current table, 
+  # 2) initialise retry counter, 
+  # 3) set max number of retry attempts, 
+  # 4) flag to track if request was successful
+  for (i in seq_along(async_api_calls)) {
+    url <- async_api_calls[i]
+    attempt <- 1
+    max_attempts <- 5
```

Comment:
Looks like a great fix, what do you think about adding `max_attempts` as an argument to the `get_nhsbsa_data` function? Or do we not need to option to change this easily when calling the function?

Reply:
Sounds good!